File: TT3.1/NameTester.pas
program NameTester;
uses TerminalUserInput;

procedure OutputSillyName(firstName: string);
var
    i: Integer;
begin
    i := 0;

    Write(firstName, ' is a');

    while i < 60 do
    begin
        Write(' silly');
        i := i + 1;
    end;

    Write(' name!');
    WriteLn();
end;

procedure Main();
var
    firstName: String;
begin
    firstName := ReadString('Please enter your name: ');

    if (lowercase(firstName) = 'dylan') then
        WriteLn(firstName, ' is an awesome name!')
    else if (lowercase(firstName) = 'marcus') then
        WriteLn(firstName, ' is an ok name!')
    else
        // WriteLn(name, ' is a silly name!');
        OutputSillyName(firstName);
end;

begin
    Main();
end.

File: TT9.1/name_tester.c
#include <stdio.h>
#include <string.h>
#include "terminal_user_input.h"

#define LOOP_COUNT 60

int main() {
    my_string name;
    int i;

    name = read_string("What is your name? ");

    if (strcmp(name.str, "Pham") == 0) {
        printf("%s, is an awesome name!\n", name.str);
    }
    else if (strcmp(name.str, "Dylan") == 0) {
        printf("Hello, Dylan\n");
    }
    else {
        printf("%s is a ", name.str);

        for (i = 0; i < LOOP_COUNT; ++i) {
            printf("silly, ");
        }
        printf("name!\n");
    }

    return 0;
}
